src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -52,9 +53,13 @@ public AddCommand parse(String args) throws ParseException {
         Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
         Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
         Description description = ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION).get());
-        NextOfKin nextOfKin = ParserUtil.parseNextOfKin(argMultimap.getValue(PREFIX_NOK).get());
         Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
 
+        Optional<NextOfKin> nextOfKin = Optional.ofNullable(null);
+        if (argMultimap.getValue(PREFIX_NOK).isPresent()) {
+            nextOfKin = Optional.of(ParserUtil.parseNextOfKin(argMultimap.getValue(PREFIX_NOK).get()));
```

Comment:
Look at previous PR review, you can replace this with neater Optional map

Reply:
If I use map like that I have to handle some parse exception.
Tried with ifPresent and map and having the same issue
<img width="963" alt="image" src="https://github.com/AY2324S2-CS2103T-W12-4/tp/assets/48405651/26cdf14f-3ef1-429e-82eb-dc5e9c840b20">

Reply:
yeah, I guess theres no easy fix for this one,